lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php

```diff
@@ -2798,24 +2775,7 @@ public function addIndex(
 
         $query .= sprintf(' ADD %s (%s)', $condition, $fieldSql);
 
-        $cycle = true;
-        while ($cycle === true) {
-            try {
-                $result = $this->rawQuery($query);
-                $cycle  = false;
-            } catch (\Exception $e) {
-                if (in_array(strtolower($indexType), ['primary', 'unique'])) {
-                    $match = [];
-                    // phpstan:ignore
-                    if (preg_match('#SQLSTATE\[23000\]: [^:]+: 1062[^\']+\'([\d.-]+)\'#', $e->getMessage(), $match)) {
-                        $ids = explode('-', $match[1]);
-                        $this->_removeDuplicateEntry($tableName, $fields, $ids);
-                        continue;
-                    }
-                }
-                throw $e;
-            }
-        }
+        $result = $this->rawQuery($query);
```

Comment:
Hi @kandy,
Could you explain whis part it was removed?
It looks not related to changes from this PR.

Reply:
I removed this code as it does not make sense. It removes not unique data from the table in case you add uniq key.
For example, if you add SKU key to product table it remove products that have not uniq sku. I don't think that it is expected behavior